### ISSUE TITLE
XCode 12 linking brocken for arm64

### ISSUE THREAD

**Issue by KirioXX** [heap/react-native-heap#230]
With the update to XCode 12 and react-native 0.63.3, the auto-linking for the arm64 architecture seems to be broken. 
We get now this error every time we try to build:
```sh
ld: in /Users/.../node_modules/@heap/react-native-heap/ios/Vendor/libHeap.a(HITelemetry.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/.../node_modules/@heap/react-native-heap/ios/Vendor/libHeap.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
As a workarround, the solution mentioned in this issue https://github.com/facebook/react-native/issues/29984#issuecomment-696328815 is working. <br/> But it would great if this could be fixed in react-native-heap to make it possible to support the arm64 architecture again.

Thank you.

**Comment by alanmf** [comment]
I ran into this issue today too after upgrading Xcode and several libraries, the workaround above worked, but I concur it would be nice to have this fixed.

**Comment by bnickel** [comment]
This is fixed in 0.17.0.  The linked heap-ios-sdk now ships as a dynamic xcframework with separate frameworks for simulators and and devices.

**Comment by Emex4gman** [comment]
I ran into the same issue building a production copy 
- XCode 13
- "@heap/react-native-heap": "^0.22.2",
-  "react-native": "0.64.3",
- Expo 44 (ejected)
- `eas build`

I tried the workaround and it didn't work for me. any one having same issue 
```
❌  ld: '/Users/expo/Library/Developer/Xcode/DerivedData/Bitbuy-
cazuuwgjttvnvlhgtcmhyrvebqsg/Build/Intermediates.noindex/ArchiveIntermediates/Bitbuy/BuildProductsPath/Release-
iphoneos/XCFrameworkIntermediates/Heap/Heap.framework/Heap' does not contain bitcode. You must rebuild it with 
bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, 
or disable bitcode for this target. file '/Users/expo/Library/Developer/Xcode/DerivedData/Bitbuy-
cazuuwgjttvnvlhgtcmhyrvebqsg/Build/Intermediates.noindex/ArchiveIntermediates/Bitbuy/BuildProductsPath/Release-
iphoneos/XCFrameworkIntermediates/Heap/Heap.framework/Heap' for architecture arm64

```
